gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/LocalGcdHelper.java

```diff
@@ -94,6 +96,17 @@
     }
   }
 
+  public static int findOpenPort() {
+    int port;
+    try (ServerSocket temp_socket = new ServerSocket(0)) {
+      port = temp_socket.getLocalPort();
+      temp_socket.close();
```

Comment:
temp_socket is going to be closed by the try (as you are using try with resource).